types/chai-as-promised/index.d.ts

```diff
@@ -32,7 +32,7 @@ declare namespace Chai {
     // Eventually does not have .then(), but PromisedAssertion have.
     interface Eventually extends PromisedLanguageChains, PromisedNumericComparison, PromisedTypeComparison {
         // From chai-as-promised
-        become(expected: PromiseLike<any>): PromisedAssertion;
+        become(expected: any): PromisedAssertion;
```

Comment:
maybe the ideal here is `Promise<any> | any`?

Reply:
It looks like `become` is essentially a synonym for `deep.equal`, and the type signature of anything `equal`-like is just `any`: https://github.com/lazerwalker/DefinitelyTyped/blob/bfa639858a16e1a71fbfd1b871d328df558bb250/types/chai-as-promised/index.d.ts#L204-L211

It also conceptually makes sense — if `expected` can be a promise,you'd be asserting that a promise's value will eventually become the value of a different promise. Which is... maybe useful? But doesn't seem common enough to explicitly call that out in the type signature.

Reply:
Ah, yeah, looked at the actual source. 

All these things get wrapped in a promise, and on resolution the expected value arg just gets passed through to the appropriate regular Chai assertion (https://github.com/domenic/chai-as-promised/blob/master/lib/chai-as-promised.js#L331-L354).

If we pass in a promise for `expected`, it's not gonna get resolved, just passed through to regular old Chai `assert`. So `any` is definitely it 👍 .